azavea/loam
# Validate defaultPrefix is a valid URL

## Overview

Validate `defaultPrefix` is a valid URL. If it isn't, use undefined instead. Using an invalid (but non-undefined) URL will always cause `loam.initialize()` to throw an exception, even if the user attempts to provide a valid URL as a parameter to initialize(). This now validates `defaultPrefix` before attempting to use it, and uses `undefined` if `defaultPrefix` is not a valid URL. The user in this case will be _required_ to provide a valid absolute URL parameter to `initialize()`, because there is no default to fall back on.

### Demo
N/A

## Testing Instructions

 * The unit tests for this project are not really structured to test this case and I couldn't find a way to test this through the regular means without a significant refactor of the testing setup. Luckily, this is a fairly straightforward change, so the best testing is to confirm the desired behavior in a console:
   * Confirm the bug: `new URL('http://www.a-valid-url.com/', 'any invalid url')` will raise an error
   * Confirm the validation: `new URL('any invalid url')` should raise an error
   * Confirm the fix `new URL('http://www.a-valid-url.com/', undefined)` should _not_ raise an error
 * Unfortunately this means that there's no way to be sure this issue won't re-occur, but it's in a low-churn area of the code and the fix is pretty explicit, so I think it's relatively unlikely to happen.
 * In the long term, enabling Node support and switching to Typescript will probably require changes that will make testing this easier.

## Checklist


Resolves #86
